File: ProjectTest/OneLang/Python/OneLang/One/Transforms/ResolveImports.py
from OneLangStdLib import *
import OneLang.One.Ast.Types as types

class ResolveImports:
    def __init__(self):
        pass
    
    @classmethod
    def process_workspace(cls, ws):
        for pkg in ws.packages.values():
            for file in pkg.files.values():
                for imp in file.imports:
                    imp_pkg = ws.get_package(imp.export_scope.package_name)
                    scope = imp_pkg.get_exported_scope(imp.export_scope.scope_name)
                    imp.imports = scope.get_all_exports() if imp.import_all else list(map(lambda x: scope.get_export(x.name) if isinstance(x, types.UnresolvedImport) else x, imp.imports))
                    file.add_available_symbols(imp.imports)

File: ProjectTest/OneLang/Java/src/main/java/OneLang/One/Transforms/ResolveImports/ResolveImports.java
import java.util.Arrays;

public class ResolveImports {
    public static void processWorkspace(Workspace ws) {
        for (var pkg : ws.packages.values().toArray(Package[]::new))
            for (var file : pkg.files.values().toArray(SourceFile[]::new))
                for (var imp : file.imports) {
                    var impPkg = ws.getPackage(imp.exportScope.packageName);
                    var scope = impPkg.getExportedScope(imp.exportScope.scopeName);
                    imp.imports = imp.importAll ? scope.getAllExports() : Arrays.stream(imp.imports).map(x -> x instanceof UnresolvedImport ? scope.getExport(((UnresolvedImport)x).getName()) : x).toArray(IImportable[]::new);
                    file.addAvailableSymbols(imp.imports);
                }
    }
}